plugins/gatsby-transformer-garden-svg/gatsby-node.js

```diff
@@ -5,9 +5,89 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+const fetch = require('node-fetch');
 const { createNodeHelpers } = require('gatsby-node-helpers');
 const { optimize } = require('svgo');
 
+const tokens = new Map();
+
+exports.onPreInit = async (_, configOptions) => {
+  const { figmaApiToken, fileId, nodeId } = configOptions;
+  const baseURL = 'https://api.figma.com/v1';
+  const headers = {
+    'Content-Type': 'application/json',
+    'X-FIGMA-TOKEN': figmaApiToken
+  };
+
+  const file = await fetch(`${baseURL}/files/${fileId}/nodes?ids=${nodeId}`, { headers });
+  const { nodes } = await file.json();
+  const frame = nodes[Object.keys(nodes)[0]];
+  const { children } = frame.document;
+
+  for (const node of children) {
+    if (node.type === 'COMPONENT') {
+      // eslint-disable-next-line prefer-named-capture-group
+      const name = node.name.replace(/\b(\s+token)\b/u, '').toLowerCase();
+      const id = node.id;
+      const iconId = node.children[0].componentId;
+      const icon = node.children[0].name
+        .split('-')[0]
+        .trim()
+        .replace('_', '')
+        .replace(/\s/gu, '-')
+        .toLowerCase();
```

Comment:
It would be good to preface this parsing block with a comment that defines the expected format of the data and the particular information that this is attempting to extract.

Have you tested this against malformatted data strings? How is this expected to gracefully fail?

Reply:
Mb, I didn't test this against malformatted data strings. I'm going to add additional checks to ensure during build time the whole application doesn't explode. I'm also going to see if I can minimize the amount of split and replace function through the use of regex.